src/app/services/file-upload.service.ts

```diff
@@ -0,0 +1,168 @@
+import { AppConfigService } from "src/app/app-config.service";
+import { HttpClient, HttpHeaders } from "@angular/common/http";
+import { Injectable, Injector } from "@angular/core";
+import { Observable, finalize, of, switchMap, tap } from "rxjs";
+import { MaybeUndefined, UploadAvailableMethod, UploadPerareData, UploadPrepareResponse } from "../common/app.types";
+import { LocalStorageService } from "./local-storage.service";
+import { DatasetInfo } from "../interface/navigation.interface";
+import { DatasetBasicsFragment } from "../api/kamu.graphql.interface";
+import { APOLLO_OPTIONS } from "apollo-angular";
+import { DatasetApi } from "../api/dataset.api";
+import { DatasetViewTypeEnum } from "../dataset-view/dataset-view.interface";
+import { DatasetService } from "../dataset-view/dataset.service";
+import { NavigationService } from "./navigation.service";
+
+@Injectable({
+    providedIn: "root",
+})
+export class FileUploadService {
+    private authHeaders: MaybeUndefined<{
+        [header: string]: string | string[];
+    }>;
+    private uploadToken: string;
```

Comment:
The headers are defined in constructor, but not the token

Reply:
Removed 2 variables from service